src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java

```diff
@@ -535,27 +544,18 @@ private MimeMessage createMail(ExtendedEmailPublisherContext context) throws Mes
             debug(context.getListener().getLogger(), "NOT overriding default server settings, using Mailer to create session");
             session = Mailer.descriptor().createSession();
             msg = new MimeMessage(session);
-            msg.setFrom(new InternetAddress(JenkinsLocationConfiguration.get().getAdminAddress()));
+            msg.setFrom(Mailer.StringToAddress(JenkinsLocationConfiguration.get().getAdminAddress(), charset));
```

Comment:
Then I can re-implement a method like `Mailer.StringToAddress` as a private method. I think it's not good way. How do you think?